storage/zynq7000-flash/zynq7000-flash.c

```diff
@@ -0,0 +1,529 @@
+	}
+	else {
+		res = -EINVAL;
+	}
+
+	return res;
+}
+
+
+static int flash_sync(oid_t *oid)
+{
+	ssize_t res = -ENOSYS;
+	storage_t *strg = storage_get(GET_STORAGE_ID(oid->id));
+
+	if (strg == NULL || strg->dev == NULL) {
+		res = -EINVAL;
+	}
+	else if (GET_MTD_TYPE(oid->id) == MTD_CHAR) {
+		res = -ENOSYS;
+	}
+	else if (strg->dev->blk != NULL && strg->dev->blk->ops != NULL && strg->dev->blk->ops->sync != NULL && GET_MTD_TYPE(oid->id) == MTD_BLOCK) {
+		res = strg->dev->blk->ops->sync(strg);
+	}
+	else {
+		res = -EINVAL;
+	}
+
+	return res;
+}
+
+
+static int flash_getAttr(oid_t *oid, int type, long long *attr)
+{
+	storage_t *strg = storage_get(GET_STORAGE_ID(oid->id));
+
+	if (strg == NULL) {
+		return -EINVAL;
+	}
+
+	switch (type) {
+		case atSize:
+			*attr = strg->size;
+			break;
+
+		default:
+			return -EINVAL;
+	}
+
+	return EOK;
+}
+
+
+static int flash_devCtl(oid_t *oid)
+{
+	ssize_t res;
+
+	/* TODO: use libmtd to handle ioctl */
+	if (GET_MTD_TYPE(oid->id) == MTD_CHAR) {
+		res = -ENOSYS;
+	}
+	else if (GET_MTD_TYPE(oid->id) == MTD_BLOCK) {
+		res = -ENOSYS;
+	}
+	else {
+		res = -EINVAL;
+	}
+
+	return res;
+}
+
+
+static void flash_msgLoop(void *arg, msg_t *msg)
+{
+	mount_msg_t *mnt;
+
+	switch (msg->type) {
+		case mtOpen:
+		case mtClose:
+			msg->o.io.err = EOK;
+			break;
+
+		case mtRead:
+			msg->o.io.err = flash_read(&msg->i.io.oid, msg->i.io.offs, msg->o.data, msg->o.size);
+			break;
+
+		case mtWrite:
+			msg->o.io.err = flash_write(&msg->i.io.oid, msg->i.io.offs, msg->i.data, msg->i.size);
+			break;
+
+		case mtSync:
+			msg->o.io.err = flash_sync(&msg->i.io.oid);
+			break;
+
+		case mtGetAttr:
+			msg->o.attr.err = flash_getAttr(&msg->i.attr.oid, msg->i.attr.type, &msg->o.attr.val);
+			break;
+
+		case mtMount:
+			mnt = (mount_msg_t *)msg->i.raw;
+			msg->o.attr.err = storage_mountfs(storage_get(GET_STORAGE_ID(mnt->id)), mnt->fstype, NULL, 0, (oid_t *)msg->o.raw);
+			break;
+
+		case mtDevCtl:
+			msg->o.io.err = flash_devCtl(&msg->i.io.oid);
+			break;
+
+		default:
+			msg->o.io.err = -EINVAL;
+			break;
+	}
+}
+
+
+static void flash_help(const char *prog)
+{
+	printf("Usage: %s [options] or no args to automatically detect and initialize a NOR flash device\n", prog);
+	printf("\t-p <dev:start:size>          - register partition\n");
+	printf("\t\tdev:    device path\n");
+	printf("\t\tstart:  partition start in bytes\n");
+	printf("\t\tsize:   partition size in bytes\n");
+	printf("\t-r <dev:start:size:fs>       - mount root filesystem\n");
+	printf("\t\tdev:    device name\n");
+	printf("\t\tstart:  partition start in bytes\n");
+	printf("\t\tsize:   partition size in bytes\n");
+	printf("\t\tfs:     filesystem name\n");
+	printf("\t-h                           - print this help message\n");
+}
+
+
+static int flash_oidResolve(const char *devPath, oid_t *oid)
+{
+	int res;
+	char temp[32];
+
+	if (strncmp("/dev/", devPath, 5) != 0) {
+		return -EINVAL;
+	}
+
+	memset(temp, 0, sizeof(temp));
```

Comment:
Is it really necessary?
If so, it is better
```
	char temp[32] = { 0 };
```

Reply:
It can be removed.